lib/active_model/serializer/adapter/json_api.rb

```diff
@@ -161,15 +169,16 @@ def add_resource_relationships(attrs, serializer, options = {})
           end
         end
 
-        def add_links(options)
-          links = @hash.fetch(:links) { {} }
-          resources = serializer.instance_variable_get(:@resource)
-          @hash[:links] = add_pagination_links(links, resources, options) if is_paginated?(resources)
+        def add_top_level_links(options)
+          @hash[:links] = @top_level_links.serializable_hash if @top_level_links.present?
```

Comment:
is there any benefit to encapsulating the `options[:links]` in an object? We should be able to use it directly, no?  That's pretty much how you wrote it...